shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -182,6 +183,117 @@
   return canvases;
 }
 
+int FlutterPlatformViewsController::GetNumberOfClips(const MutatorsStack& mutators_stack) {
+  // Apply transforms/clips.
+  std::vector<std::shared_ptr<Mutator>>::const_reverse_iterator iter = mutators_stack.bottom();
+  int clipCount = 0;
+  while (iter != mutators_stack.top()) {
+    if ((*iter)->isClipType()) {
+      clipCount++;
+    }
+    ++iter;
+  }
+  return clipCount;
+}
+
+UIView* FlutterPlatformViewsController::ReconstructClipViewsChain(int number_of_clips,
+                                                                  UIView* child,
+                                                                  UIView* parent) {
+  NSInteger index = -1;
+  if (parent.superview) {
+    // TODO(cyanglaz): protentially cache the index of oldPlatformViewRoot to make this a O(1).
+    index = [flutter_view_.get().subviews indexOfObject:parent];
+    [parent removeFromSuperview];
+  }
+  UIView* head = child;
+  BOOL needAddNewView = NO;
+  for (int i = 0; i < number_of_clips; i++) {
+    if (head == parent) {
```

Comment:
Assuming we don't need `needAddNewView`(see comment below) maybe it's simpler to:

```dart
if (head != parent) {
  head = head.superview;
  continue;
}
```